### DISCUSSION TITLE
Significant release for HERO12 today (December '23)

### DISCUSSION THREAD

**Discussion by dnewman-gpsw** [gopro/labs#683] [Announcements]
V2.0 firmware is out, and Labs gains many features too

### NEW FEATURES (Mainline)
- Enables renaming of custom presets using the GoPro Quik app.
- Adds HyperSmooth AutoBoost option when using Max Lens Mod 2.0.
- Adds Linear lens option when using Max Lens Mod 2.0.
- Adds Max HyperView lens option for TimeWarp when using Max Lens Mod 2.0.

### IMAGE QUALITY IMPROVEMENT
- Optimizes GP-Log tuning with wide-gamut color space.

 ### GOPRO LABS CHANGES
- Added FEAT - Display the Labs feature you have active, without have to reboot the camera. e.g. oMFEAT=4 <- display for 4 seconds.
- Added creating, naming and sharing presets oMPRES=”1,MyPreset” 0-18 for icon selection, -1 for delete
- Added FONT=1 Select the old fonts (if you only want the dot matrix font)
- Added GRAB=x screen grab x-frames, !F (grab now) and !F3 (grab next 3 Labs overlays) - frame grab the UI. Useful for education and bug reporting
- Added font color for script messaging. e.g. try this QR Code “\1H\2e\3l\4l\5o \6W\7o\8r\1l\2d”
- Improved the font rendering (higher res fonts)
- Improved support for Max Lens Mod 2.0
- Improved upload reliability with an optional maximum transfer time (timeout). e.g. !U3 <- upload now, for up to 3 minutes.

**Key Labs Feature demos**
Feature display via QR Code using oMFEAT=1
![image](https://github.com/gopro/labs/assets/10856084/e9ff802d-ce0f-495e-a66b-6f38e589ad59)
This also demo the new screen grab to JPG that will help in debugging and sharing Labs idea.  For the above I used the command !F1oMFEAT=4 which set the frame grab to capture the next Labs screen message. 

Sharing named presets. 
On the main QR Code Creator page there is now name preset creating option:
![image](https://github.com/gopro/labs/assets/10856084/bdbcdcbc-b56a-4c52-9975-1c503fd5646e)
So you can create shareable/branded QR Code like this:  
![image](https://github.com/gopro/labs/assets/10856084/c4f979f8-f73e-45a7-9322-d140b2253e05)
The Preset names need to 16 characters or less.

**Comment by tomislav** [comment]
Does “Optimizes GP-Log tuning with wide-gamut color space” mean WIDE=1 is on by default?

**Reply by dnewman-gpsw** [reply]
Correct. GP-Log is equivalent to log base 400 with wide enabled now.  For Labs uses, I would still recommend combining GP-Log with !MNR01=50 or lower.  While Labs has been used to prototype the GP-Log feature, helping it go mainstream, there are still more imaging controls in Labs to get the best in performance.

**Reply by RobGoniacz** [reply]
Hello, I'm quite new in GoPro world, and I've a few questions regarding Labs FW for GoPro HERO 12. Whether the LOGB=400 setting passing as the QR code completely replaces curves for Standard and HDR profiles ? When I did it (set LOGB=400 via QR code) my Standard and HDR settings/profiles were useless. They are full or chroma noise, too bright with big color spots - as if they were pushed in ISO sensitivity. They are back to normal when I set LOGB=113 (!) via QR code. Is it possible to replace curves for Standard, HDR and Log separately ? Regards, Robert

**Reply by dnewman-gpsw** [reply]
LOGB will replace curves globally on the camera. It can't be used with HDR. GoPro HDR is about a developed tone-mapped, great looking image in-camera, not about a flat image for post.  To restore default curves use !MLOGB=0.  On HERO12 GP-Log is LOGB=400, so you only the LOGB hack is you using Max Lens Mode 2.0 which doesn't offer GP-Log.  When using any enhanced Labs curve, I would recommend some negative EV.  Log will make thing much brighter, emphasizing, shadows and noise (nature of Log.)  

GP-Log on HERO12 is oMWIDE=1oMLOGB=400.  Most Labs used add at least oMNR01=50oMBITR=160 or similar. As you want more noise/detail, and more bitrate to have to best chances to enhance in post.   This user has dialed in great Labs settings https://www.youtube.com/watch?v=mpsJwWL-HMw 

P.S. This subject is not related to the HERO12 firmware release notice. Should use a new discussion thread next time.

**Reply by RobGoniacz** [reply]
Thanks. Regards, Robert

**Comment by hwy0010** [comment]
Can we except same labs feature update for HERO11?

**Reply by dnewman-gpsw** [reply]
Not all the new features can run on H11.  There is more coming for HERO11, next year.
